libiocage/lib/Release.py

```diff
@@ -310,75 +369,129 @@ def update(self):
+                ),
+                logger=self.logger
+            )
+
+        jail.stop()
+
+        self.logger.verbose(f"Release '{self.name}' updated")
+        return changed
+
+    def _update_freebsd_jail(self, jail):
+
         local_update_mountpoint = f"{self.root_dir}/var/db/freebsd-update"
         if not os.path.isdir(local_update_mountpoint):
             self.logger.spam(
                 "Creating mountpoint {local_update_mountpoint}"
             )
             os.makedirs(local_update_mountpoint)
 
-        try:
+        jail.config.fstab.add(
+            self.release_updates_dir,
+            local_update_mountpoint,
+            "nullfs",
+            "rw"
+        )
+        jail.config.fstab.save()
 
-            jail.config.fstab.add(
-                self.release_updates_dir,
-                local_update_mountpoint,
-                "nullfs",
-                "rw"
-            )
-            jail.config.fstab.save()
-
-            jail.start()
-
-            child, stdout, stderr = jail.exec([
-                "/var/db/freebsd-update/freebsd-update.sh",
-                "-d",
-                "/var/db/freebsd-update",
-                "-f",
-                "/var/db/freebsd-update/freebsd-update.conf",
-                "install"
-            ], ignore_error=True)
-
-            if child.returncode == 1:
-                if "No updates are available to install." in stdout:
-                    self.logger.debug("Already up to date")
-                    changed = True
-                else:
-                    raise libiocage.lib.errors.ReleaseUpdateFailure(
-                        reason=(
-                            "freebsd-update.sh exited "
-                            "with returncode {child.returncode}"
-                        ),
-                        logger=self.logger
-                    )
-            else:
-                self.logger.debug(f"Update of release '{self.name}' finished")
+        jail.start()
+
+        child, stdout, stderr = jail.exec([
+            "/var/db/freebsd-update/freebsd-update.sh",
+            "-d",
+            "/var/db/freebsd-update",
+            "-f",
+            "/var/db/freebsd-update/freebsd-update.conf",
+            "install"
+        ], ignore_error=True, logger=self.logger)
+
+        if child.returncode == 1:
+            if "No updates are available to install." in stdout:
+                self.logger.debug("Already up to date")
                 changed = True
+            else:
+                raise libiocage.lib.errors.ReleaseUpdateFailure(
+                    release_name=self.name,
+                    reason=(
+                        "freebsd-update.sh exited "
+                        f"with returncode {child.returncode}"
+                    ),
+                    logger=self.logger
+                )
+        else:
+            self.logger.debug(f"Update of release '{self.name}' finished")
+            changed = True
```

Comment:
Same here, why set `changed` at all? It can only ever be true or we raise an exception.